### PR TITLE
fix: SteamOpenId does not validate identity url

### PR DESCRIPTION
**Pull request by zhuang42**
## Proposed changes

The implementation of SteamOpenId does not validate identity URL and could be impersonated by attackers.

## Types of changes

Please check the type of change your PR introduces:

- [ ] Release (new release request)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (PEP8, lint, formatting, renaming, etc)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes (build process, tests runner, etc)
- [ ] Other (please describe):

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask. We're here to
help! This is simply a reminder of what we are going to look for before merging
your code._

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works

## Other information

Any other information that is important to this PR such as screenshots of how
the component looks before and after the change.

### CODE REVIEW

**Comment by zhuang42** [comment]
#538

**Comment by nijel** [comment]
Anyone tried my suggestion in https://github.com/python-social-auth/social-core/issues/538#issuecomment-787514783 instead? This patch IMHO doesn't address https://github.com/python-social-auth/social-core/issues/538 at all.

**Comment by zhuang42** [comment]
The Steam openId login can still be stateless. Also I don't see how adding stores in https://github.com/python-social-auth/social-core/issues/538#issuecomment-787514783 address the problem, since it doesn't validate the op_endpoint/claim_id from the response still. Could you explain more about your fix?

**Comment by nijel** [comment]
If I understood https://github.com/python-social-auth/social-core/issues/538 correctly, the identify URL can be spoofed by third-party server. Restricting it to spoofing valid URL doesn't IMHO address the issue. What we really need is validating that the assertion is coming from the correct server, and keeping state should ensure that. (I might be wrong, my OpenID knowledge is very limited)

File: social_core/backends/steam.py [hunk 1]
```diff
@@ -45,6 +45,8 @@ def consumer(self):
         return self._consumer
 
     def _user_id(self, response):
+        if not response.identity_url.startswith(SteamOpenId.URL):
```
Comment (nijel) [inline social_core/backends/steam.py:48]:
```suggestion
        if not response.identity_url.startswith(self.URL):
```

**Comment by nijel** [comment]
Merged as this does not harm, but I'm still not convinced this really addresses the original issue.

**Pull request merged**